connect/runtime/src/test/java/org/apache/kafka/connect/runtime/ExactlyOnceWorkerSourceTaskTest.java

```diff
@@ -687,7 +687,26 @@ public void testCommitFlushCallbackFailure() throws Exception {
             callback.onCompletion(failure, null);
             return null;
         });
-        testCommitFailure(failure);
+        testCommitFailure(failure, false);
+    }
+
+    @Test
+    public void testCommitFlushAsyncCallbackFailure() throws Exception {
+        Exception failure = new RecordTooLargeException();
+        when(offsetWriter.willFlush()).thenReturn(true);
+        when(offsetWriter.beginFlush()).thenReturn(true);
+        // doFlush delegates it's callback to the producer,
+        // which delays completing the callback until commitTransaction
+        AtomicReference<Callback<Void>> callback = new AtomicReference<>();
+        when(offsetWriter.doFlush(any())).thenAnswer(invocation -> {
+            callback.set(invocation.getArgument(0));
+            return null;
+        });
+        doAnswer(invocation -> {
+            callback.get().onCompletion(failure, null);
+            throw failure;
```

Comment:
We already have test coverage for the case where `producer::commitTransaction` throws an exception. Just to be sure that we're testing the offset flush callback instead of the try/catch around the transaction commit, we shouldn't throw anything from here.

```suggestion
            return null;
```

Reply:
this makes a lot of sense, thanks for catching this!